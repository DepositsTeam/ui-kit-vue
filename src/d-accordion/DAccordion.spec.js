import { mount } from "@vue/test-utils";
import DAccordion from "./DAccordion.vue";
import { expect, test } from "vitest";

test("renders the component", async () => {
  expect(DAccordion).to.not.be.undefined;

  const wrapper = mount(DAccordion, {
    props: {
      title: "Personal information",
    },
  });

  expect(wrapper.get(".ui-accordion__header__title").text()).to.be.equal(
    "Personal information"
  );

  expect(wrapper.find('[data-test="title-text"]').exists()).to.be.true;
});

test("matches snapshot", async () => {
  const wrapper = mount(DAccordion);

  expect(wrapper.element).toMatchSnapshot();
});

test("toggles the accordion on click", async () => {
  const wrapper = mount(DAccordion, {
    props: {
      title: "Personal information",
    },
    slots: {
      default: "I am the expanded content",
    },
  });

  expect(wrapper.find(".ui-accordion__content").exists()).to.be.false;

  expect(wrapper.get(".ui-accordion__header").classes("active")).to.be.false;

  await wrapper.get(".ui-accordion__header").trigger("click");

  const emitted = wrapper.emitted("toggled-accordion");

  expect(emitted).to.have.lengthOf(1);

  expect(emitted[0][0]).to.be.true;

  expect(wrapper.get(".ui-accordion__content").text()).to.be.equal(
    "I am the expanded content"
  );

  expect(wrapper.get(".ui-accordion__header").classes("active")).to.be.true;

  await wrapper.get(".ui-accordion__header").trigger("click");

  expect(wrapper.find(".ui-accordion__content").exists()).to.be.false;

  expect(wrapper.get(".ui-accordion__header").classes("active")).to.be.false;

  expect(emitted).to.have.lengthOf(2);

  expect(emitted[1][0]).to.be.false;
});

test("slots work as expected", async () => {
  const wrapper = mount(DAccordion, {
    slots: {
      default: "I am the expanded content",
      title: "I am a slot title",
    },
  });
  expect(wrapper.find('[data-test="title-text"]').exists()).to.be.false;
  expect(wrapper.get(".ui-accordion__header__title").text()).to.be.equal(
    "I am a slot title"
  );
});
