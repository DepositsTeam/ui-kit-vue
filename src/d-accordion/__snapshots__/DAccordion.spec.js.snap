// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div
  class="circularSTD deposits-ui-box ui-accordion"
  data-ui-kit-auto_generated-cltvdguj3s=""
  data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
  data-v-16297268=""
>
  
  <div
    class="circularSTD deposits-ui-box ui-accordion__header"
    data-ui-kit-auto_generated-cltvdguj3s=""
    data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
    data-v-16297268=""
  >
    
    <div
      class="circularSTD deposits-ui-box ui-accordion__header__left-content"
      data-ui-kit-auto_generated-cltvdguj3s=""
      data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
      data-v-16297268=""
    >
      
      <div
        class="circularSTD deposits-ui-box ui-accordion__header__left-icon"
        data-ui-kit-auto_generated-cltvdguj3s=""
        data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
        data-v-16297268=""
      >
        
        
        
        <div
          class="circularSTD deposits-ui-box text-grey-500"
          data-ui-kit-auto_generated-cltvdguj3s=""
          data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
          data-v-16297268=""
        >
          
          <svg
            class="deposits-ui-box"
            data-ui-kit-auto_generated-FYeTYoeo4j=""
            data-ui-kit-theming-rules-auto_generated-FYeTYoeo4j=""
            data-v-16297268=""
            fill="none"
            focusable="false"
            height="32px"
            viewBox="0 0 24 24"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
          >
            
            <path
              d="M7.10108 9.65718L11.7577 14.8872C11.8915 15.0376 12.1086 15.0376 12.2424 14.8872L16.8989 9.65718C17.1148 9.41466 16.962 9 16.6565 9H7.34344C7.03808 9 6.88516 9.41467 7.10108 9.65718Z"
              fill="currentcolor"
            />
            
          </svg>
          
        </div>
        
        
        
      </div>
      <div
        class="circularSTD deposits-ui-box ui-accordion__header__title"
        data-ui-kit-auto_generated-cltvdguj3s=""
        data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
        data-v-16297268=""
      >
        
        
      </div>
      
    </div>
    <div
      class="circularSTD deposits-ui-box ui-accordion__header__right-icon"
      data-ui-kit-auto_generated-cltvdguj3s=""
      data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
      data-v-16297268=""
    >
      
      
    </div>
    
  </div>
  <!--v-if-->
  
</div>
`;
